core/include/kernel/tee_common_otp.h

```diff
@@ -13,6 +13,8 @@
 
 struct tee_hw_unique_key {
 	uint8_t data[HW_UNIQUE_KEY_LENGTH];
+	/* The hardware unique key is now inmmutable */
+	bool locked;
```

Comment:
How does this relate to `plat_rpmb_key_is_ready()`?

Reply:
`plat_rpmb_key_is_ready` often depends on the platform being considered closed/secured  - ie, the bootrom booting signed firmware. However it should also depend on the HUK key being immutable (so the RPMB derived key doesnt change on a subsequent boot causing the RPMB not to be accesible). 

So IMO having some information with respect to the RPMB key derivation mutability is required. I propose to do it here since the only reason for variability in the RPMB key is the HUK  changing,  but really what plat_rpmb_key_is_ready needs to know is if the  key that has been generated can change; then it can decide whether to allow the key to be written or not.

Reply:
I agree with Jens. It's up to the platform to refuse or not to provide a HUK. There is no unique definition of what is a locked/closed device, possibly several levels of device closure.